app/src/common/shared/com/igalia/wolvic/downloads/DownloadsManager.java

```diff
@@ -134,6 +146,69 @@ public void startDownload(@NonNull DownloadJob job) {
         }
     }
 
+    public void downloadBlobUri(DownloadJob job) {
+        if (!UrlUtils.isBlobUri(job.getUri()) || job.getInputStream() == null) {
+            Log.w(LOGTAG, "Failed to download Blob URI: " + job.getUri());
+            return;
+        }
+
+        final File dir = mContext.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS);
+        if (dir == null || (dir.exists() && !dir.isDirectory())) {
+            Log.e(LOGTAG, "Error when saving " + job.getUri() + " : failed to get the Downloads directory");
+            return;
+        }
+
+        File file = new File(dir, job.getFilename());
+        if (file.exists()) {
+            // If the file already exists, we try to generate a new one.
+            String extension = MimeTypeMap.getFileExtensionFromUrl(file.toString());
+            if (!StringUtils.isEmpty(extension)) {
+                extension = '.' + extension;
+            }
+            String name = file.getName();
+            int lastDot = name.lastIndexOf('.');
+            if (lastDot >= 0) {
+                name = name.substring(0, name.lastIndexOf('.'));
```

Comment:
You can use `lastDot` here. Also perhaps rename it to lastDotPosition or lastDotIndex

Reply:
Done. I renamed the variables to `lastDotIndex` and `lastDashIndex`.